backend/controllers/userController.js

```diff
@@ -0,0 +1,66 @@
+const asyncHandler = require("express-async-handler")
+const generateToken = require("../utils/generateToken")
+const { ErrorHandler } = require("../middleware/errorMiddleware")
+const User = require("../model/user.model")
+
+//@desc Register a new user
+//@route POST /users/registerUser
+
+const registerUser = asyncHandler(async (req, res, next) => {
+  try {
+    const { name, email, password, confirmpwd, phone } = req.body
+
+    const userExists = await User.findOne({ email })
+
+    if (userExists) {
+      throw new ErrorHandler(403, "User already exists, try login")
+    }
+    if (password != confirmpwd) {
```

Comment:
Can we check this condition before line 13 because if the password are mismatched there is no sense to check whether user exists or not?

Reply:
yes, seems accurate. Done